src/App.jsx

```diff
@@ -1,28 +1,91 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+import EmojiPicker from 'emoji-picker-react';
 import './App.css';
 
 const initialColumns = {
   todo: {
     id: 'todo',
     title: 'To Do',
-    tasks: []
-  },
-  inProgress: {
-    id: 'inProgress',
-    title: 'In Progress',
-    tasks: []
-  },
-  done: {
-    id: 'done',
-    title: 'Done',
-    tasks: []
+    tasks: [],
+    order: 0,
+    emoji: '📝',
+    color: '#1a73e8'
   }
 };
 
 function App() {
   const [columns, setColumns] = useState(initialColumns);
   const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [boardTitle, setBoardTitle] = useState('Kanban Board');
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const titleInputRef = useRef(null);
+  const [showEmojiPicker, setShowEmojiPicker] = useState(null);
+  const [showColorPicker, setShowColorPicker] = useState(null);
+  const emojiPickerRef = useRef(null);
+
+  useEffect(() => {
+    if (isEditingTitle && titleInputRef.current) {
+      titleInputRef.current.focus();
+      titleInputRef.current.select();
+    }
+  }, [isEditingTitle]);
+
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (emojiPickerRef.current && !emojiPickerRef.current.contains(event.target)) {
+        setShowEmojiPicker(null);
+        setShowColorPicker(null);
+      }
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const handleDeleteColumn = (columnId) => {
  const column = columns[columnId];
  if (column.tasks.length > 0) {
    if (!window.confirm(`Are you sure you want to delete "${column.title}" and all its tasks?`)) return;
  }
  const newColumns = { ...columns };
  delete newColumns[columnId];
  setColumns(newColumns);
};
```
Add confirmation dialog before deleting a column to prevent accidental deletions, especially if the column contains tasks.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;